cli/azd/pkg/tools/github/github.go

```diff
@@ -223,12 +257,65 @@ func (cli *ghCli) GitHubActionsExists(ctx context.Context, repoSlug string) (boo
 	return true, nil
 }
 
+func (cli *ghCli) ForceConfigureAuth(authMode AuthTokenSource) {
+	if authMode == TokenSourceFile {
+		for _, tokenEnvVarName := range TokenEnvVars {
+			cli.overrideTokenEnv = append(cli.overrideTokenEnv, fmt.Sprintf("%v=", tokenEnvVarName))
+		}
+	} else { // authMode == TokenSourceEnvVar
+		// GitHub CLI will always use environment variables first.
+		// Therefore, we simply need to clear our environment context override to force environment variable usage.
+		cli.overrideTokenEnv = nil
+	}
+}
+
+func (cli *ghCli) newRunArgs(args ...string) exec.RunArgs {
+	runArgs := exec.NewRunArgs("gh", args...)
+	if cli.overrideTokenEnv != nil {
+		runArgs = runArgs.WithEnv(cli.overrideTokenEnv)
+	}
+
+	return runArgs
+}
+
+func (cli *ghCli) runAuthenticated(ctx context.Context, runArgs exec.RunArgs) (exec.RunResult, error) {
```

Comment:
do we have a `run` alone or `runNoAuth` ?   or why are we describing the way `run` works ?